liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -368,6 +381,64 @@ protected void copyConfigFiles() throws Exception {
         }
     }
 
+    //merges configured serverEnvFile with envMavenProps if specified
+    private String mergeServerEnvFileAndEnvMavenProps(String serverEnvPath) throws IOException {
+        StringBuilder updatedServerEnvPath = new StringBuilder(serverEnvPath);
+
+        File serverEnv = new File(serverDirectory, "server.env");
+        Map<String, String> serverEnvProps = convertServerEnvToProperties(serverEnv);
+
+        //merge specified server.env if present
+        if (serverEnvFile != null && serverEnvFile.exists()) {
+            Map<String, String> serverEnvFileProps = convertServerEnvToProperties(serverEnvFile);
+            serverEnvProps.putAll(serverEnvFileProps);
+        }
+
+        //merge server env props
+        if (!envMavenProps.isEmpty()) {
+            serverEnvProps.putAll(envMavenProps);
+        }
+
+        writeServerEnvProperties(serverEnv, serverEnvProps);
+        return updatedServerEnvPath.toString();
+    }
+
+    // Merge configured serverEnvFile with envMavenProps if specified
+    private void mergeConfigDirServerEnv(File configDirServerEnv, File generatedServerEnv) throws IOException {
+        Map<String, String> configDirServerEnvProps = convertServerEnvToProperties(configDirServerEnv);
+        Map<String, String> generatedServerEnvProps = convertServerEnvToProperties(generatedServerEnv);
+
+        // Merge the properties, with the configDirServerEnvProps overwriting any conflicts with generatedServerEnvProps
+        generatedServerEnvProps.putAll(configDirServerEnvProps);
+
+        writeServerEnvProperties(configDirServerEnv, generatedServerEnvProps);
```

Comment:
Wait...are you writing the merged content to the configDirServerEnv, which is in the source tree? Definitely should not do that.